src/frontends/tensorflow/src/saved_model.hpp

```diff
@@ -0,0 +1,240 @@
+    // List of opened data files for using with BundleEntryProto
+    std::map<int32_t, std::shared_ptr<std::ifstream>> m_data_files;
+    // List of mapped variables which could be read using TrackableObjectGraph
+    std::map<std::string, std::string> m_variables_map;
+
+public:
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::string& path);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::wstring& path);
+#endif
+
+    /// \brief Returns data and size of data of stored variable
+    /// \param name Name of variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_variable(const std::string& name, char** data, size_t* size) {
+        auto varItem = m_variables_index.find(name);
+        if (varItem == m_variables_index.end()) {
+            return false;
+        }
+        if (data != nullptr) {
+            *data = varItem->second.data();
+        }
+        if (size != nullptr) {
+            *size = varItem->second.size();
+        }
+        return true;
+    }
+
+    /// \brief Returns data and size of data of mapped variable from trackable object graph to variables index
+    /// \param name Name of a mapping variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_mapped_variable(const std::string& name, char** data, size_t* size) {
+        auto mapItem = m_variables_map.find(name);
+        if (mapItem == m_variables_map.end()) {
+            return false;
+        }
+        return get_variable(mapItem->second, data, size);
+    }
+
+    /// \brief Checks if variable has a mapped pair
+    /// \param name Name of variable for checking existance
+    /// \returns True in case variable has mapped value and false otherwise
+    bool has_mapped_variable(const std::string& name) {
```

Comment:
```suggestion
    bool has_mapped_variable(const std::string& name) const {
```

Reply:
very...

Reply:
this should be constant method so let us have they keyword `const` there.

Reply:
resolved?